peachjam_ml/models.py

```diff
@@ -147,6 +147,50 @@ def refresh_for_document(cls, document):
 
         return doc_embedding
 
+    @classmethod
+    def get_average_embedding(cls, pks):
+        """Get the average embedding for a set of documents."""
+        if not pks:
+            return None
+
+        if len(pks) == 1:
+            # if there's only one document, we can just return its embedding
+            doc_embedding = cls.objects.filter(document__pk=pks[0]).first()
+            if doc_embedding:
+                return doc_embedding.text_embedding
+
+        avg = (
+            DocumentEmbedding.objects.filter(document__in=pks)
+            .aggregate(avg=Avg("text_embedding"))
+            .get("avg")
+        )
+        if avg is not None and len(avg):
+            avg = normalize_vector(avg)
+
+        return avg
+
+    @classmethod
+    def get_similar_documents(cls, doc_ids, threshold=0.0):
+        # always exclude the documents we are comparing against
+        exclude_ids = doc_ids
+        avg_embedding = cls.get_average_embedding(doc_ids)
+
+        similar_docs = (
+            DocumentEmbedding.objects.exclude(document__work__documents__in=exclude_ids)
```

Comment:
I don't think this works correctly? I think you need a sub query to find the works to exclude

Reply:
this works as expected. it excludes all documents that we are using to calculate similarity

Reply:
You'll have to show me -- we need to exclude all other documents that have the same works as the documents we're looking at. We don't want to include historical versions of a legislation document, for example.